### PR TITLE
Show dependencies for casks

### PR DESCRIPTION
**Pull request by HaraldNordgren**
When showing cask dependency info, also show the installation status.

This mirrors how it already works for formulas.

### CODE REVIEW

**Comment by HaraldNordgren** [comment]
The issue that I run into so far is that when cask dependencies are just strings, so it fails when trying to check if they are satisfies in `::Utils::Info.decorate_dependencies`.

For the formula the dependencies a a fully-fledged type.

File: Library/Homebrew/utils/info.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# typed: true
+# frozen_string_literal: true
+
+module Utils
+  # Helper functions for formula and cask info.
+  module Info
+    def self.decorate_dependencies(dependencies)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/info.rb:7]: Looking at the implementation here: as you've mentioned formulae and casks don't have enough overlap here for the implementation to be sharable between formulae/casks.

Instead, I'd suggest adding a `decorate_dependencies` method to `Library/Homebrew/cask/info.rb` which takes the formula/casks as strings and then uses that to calculate dependencies, relying on `pretty_installed`/`pretty_installed` for the shared logic instead.

Make sense? 

Shout if you want more help in that direction.

**Comment by MikeMcQuaid** [comment]
@HaraldNordgren I'm game to help more if you want to keep this open?

**Comment by HaraldNordgren** [comment]
Thanks @MikeMcQuaid!

After the vacation period, I am now back on my full-time job again. Combined with having kids, this doesn't leave a ton of time to work on open source, unfortunately.

I will re-open this if I get some inspiration! 🤗

**Comment by MikeMcQuaid** [comment]
@HaraldNordgren Thanks for the heads up!

**Comment by HaraldNordgren** [comment]
Hi @MikeMcQuaid!

One year later and I'm on winter vacation so I have some time again 🎄

I think I got it to work now, do you want to take a look? 🤗

File: Library/Homebrew/utils/output.rb [hunk 2]
```diff
@@ -237,7 +237,7 @@ def pretty_outdated(string)
       sig { params(string: String).returns(String) }
       def pretty_uninstalled(string)
         if !$stdout.tty?
-          string
+          "#{string} ✘"
```
Comment (HaraldNordgren) [inline Library/Homebrew/utils/output.rb]: Without these changes, I couldn't find a way to assert my changes.

Unfortunately, this requires changes to a lot of other tests. Is there a better way?

Comment (MikeMcQuaid) [reply Library/Homebrew/utils/output.rb]: Ensure that `stdout.tty` is mocked to false. Cannot accept these changes as-is.

Comment (HaraldNordgren) [reply Library/Homebrew/utils/output.rb]: @MikeMcQuaid Is there a canonical way to achieve that? I tried a few things before and couldn't get that to work.

Comment (HaraldNordgren) [reply Library/Homebrew/utils/output.rb]: @MikeMcQuaid I found a way now, there will be a lot of control characters in the test output, is it worth it?

**Comment by HaraldNordgren** [comment]
@MikeMcQuaid Follow-up, good to have some better test coverage here: https://github.com/Homebrew/brew/pull/21294

**Pull request closed without merging**